src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp

```diff
@@ -334,18 +334,25 @@ bool FollowMovementGenerator<T>::DoUpdate(T* owner, uint32 time_diff)
 
     target->GetNearPoint(owner, x, y, z, _range, 0.f, target->ToAbsoluteAngle(tAngle));
 
+    Movement::MoveSplineInit init(owner);
+
     bool success = i_path->CalculatePath(x, y, z, forceDest);
     if (!success || i_path->GetPathType() & PATHFIND_NOPATH)
     {
-        owner->StopMoving();
-        return true;
+        // Pet must always follow its master
+        if (!followingMaster)
+        {
+            owner->StopMoving();
+            return true;
+        }
+
+        init.MoveTo(x, y, z);
```

Comment:
I was wondering: I don't  think you need to use the MoveTo here since the path is calculated in any case (as a shortcut). 

If you follow the pathgenerator logic, you can notice that it generates a shortcut but returns the PATHFIND_NOPATH flag, we can use the calculated shortcut in any case.

Therefore, you can directly add the !followingMaster inside the first condition so the creature stop to move only when it's not a pet otherwise it will use the precalculated shortcut.

It will also solve the double call that has been reported

Reply:
I don't get it. Where is that doubled call?

Reply:
after the init.MoveTo the code go ahead and then there's the .MovebyPath because you do not return anymore.

so just remove the MoveTo and simplify the condition, the code can safely go ahead and it will use the shortcut since the path is forced